src/app/plan/duck/selectors.ts

```diff
@@ -134,8 +131,41 @@ const getCounts = createSelector(
   }
 );
 
+const getPlanDiffSelector = createSelector(
```

Comment:
Optimizing rerenders using reselect memoization -> 
https://github.com/reduxjs/reselect#createselectorinputselectors--inputselectors-resultfunc

-create a selector function that compared the current and old plan objects
- create another selector function that accepts the diff selector as an argument + export this function for use in the UI. 
- if the selectors arguments do not change(in this case the returned value from diff selector function is the argument) the memoized data is passed back from the exported selector to mapstatetoprops. Which won’t trigger a rerender